api/exp/livesessions.go

```diff
@@ -98,6 +126,24 @@ func (s *AntidoteAPI) ListLiveSessions(ctx context.Context, _ *empty.Empty) (*pb
 	return &pb.LiveSessions{Items: lsAPIs}, nil
 }
 
+// UpdateLiveSessionPersistence updates the persistence flag in the session database
+func (s *AntidoteAPI) UpdateLiveSessionPersistence(ctx context.Context, persistence *pb.SessionPersistence) (*empty.Empty, error) {
+	span := ot.StartSpan("api_livesession_persist", ext.SpanKindRPCClient)
+	defer span.Finish()
+
+	_, err := s.Db.GetLiveSession(span.Context(), persistence.SessionID)
```

Comment:
UpdateLiveSessionPersistence in `db` actually performs a check for an existing livesession after acquiring a mutex lock. This is done so we don't have any weird race conditions in the time between an initial check and the update.

So, let's avoid this first call to `s.Db.GetLiveSession`, and instead just skip ahead to calling `s.Db.UpdateLiveSessionPersistence`.

Reply:
Okay, yeah in that case this check is unnecessary, i'll remove this